From e071d3986f620ece73298b7cef6a44e0b3cab944 Mon Sep 17 00:00:00 2001
From: Jani Nikula <jani.nikula@intel.com>
Date: Thu, 15 Sep 2016 16:28:53 +0300
Subject: [PATCH 0852/4301] drm/i915: do not use 'false' as a NULL pointer

commit bcbfcc37cdd4fd7564a6657d5d6bd4719079900b upstream

Fixes sparse warning:

drivers/gpu/drm/i915/intel_dpll_mgr.c:1712:24: warning: Using plain
integer as NULL pointer

Fixes: a277ca7dc01d ("drm/i915: Split bxt_ddi_pll_select()")
Cc: Manasi Navare <manasi.d.navare@intel.com>
Cc: Ander Conselvan de Oliveira <ander.conselvan.de.oliveira@intel.com>
Cc: Durgadoss R <durgadoss.r@intel.com>
Cc: Rodrigo Vivi <rodrigo.vivi@intel.com>
Reviewed-by: Dhinakaran Pandiyan <dhinakaran.pandiyan@intel.com>
Signed-off-by: Jani Nikula <jani.nikula@intel.com>
Link: http://patchwork.freedesktop.org/patch/msgid/1473946137-1931-2-git-send-email-jani.nikula@intel.com
---
 drivers/gpu/drm/i915/intel_dpll_mgr.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dpll_mgr.c b/drivers/gpu/drm/i915/intel_dpll_mgr.c
index 4b067ac..c26d18a 100644
--- a/drivers/gpu/drm/i915/intel_dpll_mgr.c
+++ b/drivers/gpu/drm/i915/intel_dpll_mgr.c
@@ -1709,12 +1709,12 @@ bxt_get_dpll(struct intel_crtc *crtc,
 	if (encoder->type == INTEL_OUTPUT_HDMI
 	    && !bxt_ddi_hdmi_pll_dividers(crtc, crtc_state,
 					  clock, &clk_div))
-		return false;
+		return NULL;
 
 	if ((encoder->type == INTEL_OUTPUT_DP ||
 	     encoder->type == INTEL_OUTPUT_EDP) &&
 	    !bxt_ddi_dp_set_dpll_hw_state(clock, &dpll_hw_state))
-		return false;
+		return NULL;
 
 	memset(&crtc_state->dpll_hw_state, 0,
 	       sizeof(crtc_state->dpll_hw_state));
-- 
1.7.5.4

